### PR TITLE
fix(ingest): four re-upload and manifest correctness refusals (#2043, #2032, #2039, #2031)

### LINKED ISSUE #2031: Reupload with a geometry-less CSV silently turns a vector dataset into a table
Observed in the 1.19.0 pre-tag smoke on the dev stack (main d39c1bbdc).

Replacing a MULTIPOINT vector dataset through the reupload path with a CSV that has no geometry columns commits without a warning: the dataset becomes a table with null geometry, and `schema_drift_status` reports nothing. The preview/commit step should surface a geometry-loss drift warning (or require an explicit confirmation) before a vector dataset loses its geometry.

Repro: upload a small point GeoJSON, then reupload a two-column CSV over it and commit.

### LINKED ISSUE #2032: srid_override accepts an unassigned EPSG code and silently ignores it
Observed in the 1.19.0 pre-tag smoke on the dev stack (main d39c1bbdc).

`srid_override` on the import/reupload commit body accepts an EPSG code that is not assigned (for example 99999) and the commit proceeds using the detected SRID, with no 422 and no note in the response. The value should be validated against the known SRIDs (spatial_ref_sys) and rejected with a 422 that names the field.

### LINKED ISSUE #2039: Manifest metadata license, organization and tags are accepted but never applied to the dataset
Observed in the 1.19.0 pre-tag smoke (main d39c1bbdc); pre-existing, not from #2021.

The v1 manifest schema accepts `metadata.license`, `metadata.organization` and `metadata.tags`, and `geolens apply` carries them into the ingest job's `user_metadata`, but `apply_manifest_record_metadata` in `backend/app/processing/ingest/tasks_common.py` copies only `manifest_attribution` onto the record (#1472). The other three fields are dropped silently, while every example under `examples/manifests/` sets them.

Ask: copy license, organization (source_organization) and tags through the same read-back, or reject them at validation time with a message that says they are not applied yet. Add a round-trip test per field.

### LINKED ISSUE #2043: Re-upload preview answers the generic 422 for a Parquet file over the ingest ceiling
Follow-up to #2036 / PR #2042.

The import preview keeps a dedicated message for `IngestBudgetExceededError` (a `.parquet` file over the ingest ceiling), while the re-upload preview now maps that error to the generic 422 "Unable to preview file. The file may be malformed or unsupported." It answered 500 before #2042, so nothing regressed, but the operator no longer learns that the ceiling, not the file, is the problem.

Ask: expose the budget error class on the catalog port (or a typed refusal) so the re-upload preview can answer the same ceiling message as the import preview, with a test for both doors.

### PR DESCRIPTION
**Pull request by ishiland**
## Summary

Four ingest/re-upload correctness fixes from the 1.19.0 follow-up triage, one commit each so they read separately. All four are server-side refusals or read-backs on the import and re-upload doors; none changes the API schema.

## Changes

- **#2043** The re-upload preview answers the Parquet ceiling message instead of the generic "malformed or unsupported" 422. `IngestBudgetExceededError` crosses `CatalogPort` (`modules/catalog/` may not import `app.processing.*`), which is a required-method addition and bumps `EXTENSION_API_VERSION` to 10.
- **#2032** An `srid_override` with no `spatial_ref_sys` row is refused with 422 naming the field and the code, at the import commit door, the re-upload commit door, and the manifest's own `metadata.crs` (so a dry run reports it). It used to commit 202 and then be dropped by the `override > detected > 4326` precedence, ingesting under the detected CRS with nothing in the response. One helper, `unknown_srid_refusal` in `app/core/geo.py`, because both doors live in different layers.
- **#2039** The manifest read-back copies `license`, `organization` (to `source_organization`) and `tags` as well as the attribution. Tags become theme keywords, skipping the ones the record already carries: a re-apply reaches the same read-back through the re-upload swap, and `uq_record_keyword` would fail that whole transaction. The accessor for the keyword ORM class is the second required port method behind the same version bump.
- **#2031** A replacement file with no geometry over a dataset that stores geometry is refused: 422 `geometry_loss` at the preview door, and the same server-authored sentence from the worker's missing-CRS gate for a client that skips the preview, so the swap that would reclassify the dataset `table` is never reached. A `table` dataset keeps accepting geometry-less replacements, and the registered-table refresh keeps its own geometry-loss handling, where an owner dropping the column is legitimate.

Two new user-facing strings (the unknown-SRID and geometry-loss refusals) are mapped in `error-map.ts` and added to all five locales; without the mapping both collapse to the generic "the submitted values are invalid".

Deviation worth flagging: #2031's commit-door refusal landed in the worker rather than in `POST .../reupload/{job}/commit`. That door holds its pooled connection from the dataset read through the run-row admission slot and the fenced CAS, and has no geometry evidence without re-running ogrinfo on the staged bytes; adding an S3 download plus a GDAL subprocess there means the #1848 rollback-and-re-read restructure on a handler carrying the #1709/#1768 fences. The worker gate refuses before the staging load, fails the job with the same message and records the refresh-run failure, so no path converts a vector dataset silently.

## Area

- [x] Backend (API, services, models)
- [x] Frontend (UI, components, hooks)
- [ ] Infrastructure (Docker, nginx)
- [ ] Tiles / Rendering
- [ ] Auth / Permissions
- [x] Import / Ingestion
- [ ] OGC / Standards
- [ ] Docs / Config

## Testing

- [x] Unit tests pass (`pytest` / `npm test`)
- [ ] Manual testing (describe below)
- [ ] Playwright e2e (if UI changes)
- [ ] No tests needed (docs, config, etc.)

Each fix has a test that fails with its source change reverted (verified per commit):

- `tests/test_reupload.py::TestReuploadPreview::test_preview_surfaces_the_ingest_ceiling_message`
- `tests/test_srid_override_validation_2032.py` (both doors, refusal and a 2263 positive control) and `tests/test_manifest_apply_service.py::TestManifestDryRun::test_dry_run_reports_a_crs_the_database_does_not_know`
- `tests/test_manifest_apply_service.py::TestManifestMetadataPropagation::{test_vector_finalize_applies_license_organization_and_tags,test_manifest_tags_are_not_reinserted_on_a_reapply}`
- `tests/test_reupload_geometry_loss_2031.py` (worker refusal plus the table case) and `test_preview_refuses_a_replacement_that_strips_geometry[vector_dataset-422]`

Also run: `ruff check` + `ruff format --check` (backend, clean); `tests/test_layering.py`, `test_rule1_structural.py`, `test_rule2_structural.py`, `test_codeql_qtable_suppressions.py`, `test_optional_auth_failure_mode_1518.py`, `test_feature_lock_order_1847.py`, `test_processing_port.py`, `test_port_signature_parity_1590.py`, `test_extensions.py`, `test_reupload.py`, `test_reupload_service.py`, `test_dataset_source_state.py`, `test_dataset_attribution.py`, `test_manifest_apply_roundtrip.py`, `test_raster_dataset_assets_bug041.py`; `backend/tests/finding_markers.py`; `scripts/dump_openapi.py --check` (snapshot unchanged, so no SDK regeneration); frontend `vitest` on `error-map` and `ReuploadDialog`, `npm run test:i18n`, `npm run lint`, `npm run typecheck`.

Four LOC caps in `test_layering.py` are ratcheted to the exact new counts with a line each saying what the growth bought.

## Checklist

- [x] Code follows existing project conventions
- [x] i18n: new user-facing strings added to all 5 locales (en, fr, es, de, zh)
- [x] No new lint warnings (`ruff check`, `eslint`)
- [x] TypeScript compiles without errors
- [ ] Migration included (if DB schema changed)
- [x] No secrets or credentials in the diff
- [ ] Release/docs deploys: ran `make deployed-surface-check` after the marketing/docs deploy and recorded the result in Testing

Closes #2043
Closes #2032
Closes #2039
Closes #2031

### CODE REVIEW

**Comment by ishiland** [comment]
@codex review

**Comment by ishiland** [comment]
@codex review

**Comment by ishiland** [comment]
@codex review

File: backend/app/processing/ingest/tasks_common.py [hunk 1]
```diff
@@ -422,7 +422,59 @@ def _parse_temporal_fields(
...
+    # fix(#2039 review): the rows `uq_record_keyword` would actually collide
+    # with — same type, same (null) vocabulary — case-folded like `wanted`. A
+    # record-wide read skipped a manifest tag that existed under another type.
+    existing = {
+        k.lower()
+        for k in await session.scalars(
+            select(RecordKeyword.keyword).where(
+                RecordKeyword.record_id == record.id,
+                RecordKeyword.keyword_type == "theme",
+                RecordKeyword.vocabulary_uri.is_(None),
```
[the remark this answers is not in the document]
Comment (ishiland) [reply backend/app/processing/ingest/tasks_common.py]: Fixed in d53a0d440: the predicate now spells the vocabulary the way uq_record_keyword does, func.coalesce(vocabulary_uri, '') == '', so a stored empty URI is seen as the collision it is. test_manifest_tags_dedupe_only_against_theme_keywords seeds a theme row with vocabulary_uri='' and fails without the change.

**Comment by ishiland** [comment]
@codex review

File: backend/app/modules/catalog/datasets/api/router_reupload.py [hunk 2]
```diff
@@ -668,6 +683,19 @@ async def reupload_preview(
                 ),
             )
 
+    # fix(#2031): the diff below reads attribute columns only, so a geometry
+    # loss reached the client as an unremarkable schema diff.
+    geometry_loss = geometry_loss_refusal(
+        record_type=prior_record_type,
+        dataset_geometry_type=prior_geometry_type,
+        source_has_geometry=info.get("geometry_type") is not None,
```
[the remark this answers is not in the document]
Comment (ishiland) [reply backend/app/modules/catalog/datasets/api/router_reupload.py:691]: Fixed in 26b2b3e06, at the parser rather than the two gates: _parse_text_ogrinfo now maps ogrinfo's 'Geometry: None' sentinel to Python None, so every caller sees the same fact — the refusal here and run_ogr2ogr's is_non_spatial derivation, which had the same latent bug. test_non_spatial_layer_reports_no_geometry in test_ingest_ogr_pure.py pins it.

**Comment by ishiland** [comment]
@codex review

File: backend/app/processing/ingest/router.py [hunk 3]
```diff
@@ -1110,6 +1111,16 @@ async def commit_import(
     # layer_name, which reaches the worker's ogr2ogr argv (see layer_guard).
     validate_commit_layer_name(job, getattr(commit, "layer_name", None))
 
+    # fix(#2032): an unassigned EPSG code committed and was then ignored.
+    srid_refusal = await unknown_srid_refusal(
+        db, getattr(commit, "srid_override", None)
+    )
```
[the remark this answers is not in the document]
Comment (ishiland) [reply backend/app/processing/ingest/router.py:1117]: Fixed in 5e718270e: unknown_srid_refusal now asks PROJ first and spatial_ref_sys second, so a code either registry knows is accepted and only a code in neither is refused. That covers the raster paths (import commit, re-upload, manifest) without a per-door raster branch; a vector job carrying a PROJ-only code now fails loudly in the worker instead of being silently ignored at the door, which is the outcome #2032 was filed about. test_a_code_only_proj_knows_is_accepted pins both halves.

File: backend/app/processing/ingest/tasks_common.py [hunk 4]
```diff
@@ -422,7 +422,60 @@ def _parse_temporal_fields(
...
+
+    fix(#2039): a re-apply reaches the reupload swap with the same list, and
+    ``uq_record_keyword`` is a unique index — an unconditional insert would
+    fail the whole swap transaction on the second apply of one manifest.
+    """
+    if not isinstance(tags, list):
+        return
+    wanted = {t.strip().lower() for t in tags if isinstance(t, str) and t.strip()}
+    if not wanted:
+        return
```
[the remark this answers is not in the document]
Comment (ishiland) [reply backend/app/processing/ingest/tasks_common.py:444]: Not fixing here, deliberately, and the lane is reporting it for a follow-up issue instead.

Reconciling would mean deleting keyword rows on every manifest apply, and nothing in the schema says which theme keywords a manifest owns: the same rows are writable by hand through the records API and by the STAC importer, which also writes keyword_type='theme'. A reconcile keyed on "theme keywords not in this manifest" would silently delete a curator's own tags on the next apply, which is a worse failure than the additive one described here.

It is also the shape the manifest read-back already has for the other three fields (#1472 attribution, and license/organization in this PR): last-write-wins per field, no deletion. Changing tags to owned-set semantics is a product decision about what a manifest owns, and it needs a marker (a vocabulary_uri, or a stored previous list) to be safe.

The symptom is real, so it goes to the orchestrator as a follow-up rather than into this PR.

**Comment by ishiland** [comment]
@codex review

File: backend/app/core/geo.py [hunk 5]
```diff
@@ -443,6 +443,52 @@ def make_bbox_filter(
...
+    fix(#2032 review): both registries count, because both consume the value:
+    a vector override reaches PostGIS ``ST_Transform``, a raster one reaches
+    ``gdal_translate -a_srs``, and the two EPSG databases are versioned
+    independently. A code in neither is refused; a code in one is the
+    consuming layer's to accept or report.
+    """
+    if srid is None:
+        return None
+    if _proj_knows_epsg(srid):
+        return None
```
[the remark this answers is not in the document]
Comment (ishiland) [reply backend/app/core/geo.py:479]: Correct on both halves, and left open for the orchestrator rather than fixed here.

The union is the second position on this code in two rounds: the prior thread on this PR asked for PROJ to be honoured because a raster override never reaches PostGIS, and this one asks for the registry to be selected per path. Per-path selection is the right end state, and it is small: the helper takes a raster flag, and each of the three doors already knows which it is (the commit subclass, the dataset record_type, the manifest prepared file_type).

The reason it is not in this PR: neither shape can corrupt data, and the two failure modes are not equal. The union accepts a divergent code and the worker then fails loudly with the real reason; the previous PostGIS-only shape refused a raster assignment the worker could have performed, which is a regression this PR would have introduced. Between the states reachable inside the review budget, this one is the safer default.

This PR is a fold-in of four filed issues; changing which registry is authoritative per ingest path is new scope on top of them. The lane is reporting it to the orchestrator with the sketch above so it lands as its own change with its own tests, or as an instruction to fold it in here.

File: backend/app/processing/ingest/tasks_common.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ishiland) [reply backend/app/processing/ingest/tasks_common.py:444]: Resolving as intended behavior. Manifest tags merge into the record's theme keywords the way every other manifest field is read back: last write wins per field and nothing is deleted. Keyword rows have no owner, the records API and the STAC importer write the same rows, so a reconcile keyed on the manifest list would delete a curator's own tags on the next apply. Removing a tag is a catalog edit. The user guide for manifests will say so.

File: backend/app/core/geo.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ishiland) [reply backend/app/core/geo.py:479]: Resolving without the per-path split. The commit door now refuses every code neither registry knows, which is what #2032 filed. A code one registry knows and the other does not is a PROJ/PostGIS version skew, and on that skew the worker still fails with the real reason instead of applying a wrong SRID, so nothing silent remains. Selecting the registry per ingest path is polish on top of that and not worth another verification round here.
